gammapy/background/ring.py

```diff
@@ -2,22 +2,236 @@
+    """
+    Adaptive ring background algorithm.
+
+    This alogrithm extends the standard `RingBackground` method by adpating the
+    size of the ring to achieve a minimum on / off exposure ratio (alpha) in regions
+    where the area to estimate the background from is limited.
+
+    Here is an illustration of the method:
+
+    Parameters
+    ----------
+    r_in : `~astropy.units.Quantity`
+        Inner radius of the ring.
+    r_out_max : `~astropy.units.Quantity`
+        Maximal outer radius of the ring.
+    width : `~astropy.units.Quantity`
+        Width of the ring.
+    stepsize : `~astropy.units.Quantity` (0.02 deg)
+        Stepsize used for increasing the radius.
+    threshold : float (0.1)
+        Threshold above which the adaptive ring takes action.
+    theta : `~astropy.units.Quantity` (0.22 deg)
+        Integration radius used for alpha computation.
+    method : {'const. width', 'const. r_in'}
+        Adaptive ring method.
+
+    Examples
+    --------
+    Here's an example how to use the `AdaptiveRingBackgroundEstimator`:
+
+    .. code::
+
+        from astropy import units as u
+        from gammapy.background import AdaptiveRingBackgroundEstimator
+        from gammapy.image import SkyImageList
+
+        filename = '$GAMMAPY_EXTRA/test_datasets/unbundled/poisson_stats_image/input_all.fits.gz'
+        images = SkyImageList.read(filename)
+        images['exposure'].name = 'exposure_on'
+        adaptive_ring_bkg = RingBackgroundEstimator(r_in=0.22 * u.deg,
+                                                    r_out_max=0.8 * u.deg,
+                                                    width=0.1 * u.deg)
+        results = adaptive_ring_bkg.run(images)
+        results['background'].show()
+
+    See Also
+    --------
+    RingBackgroundEstimator, KernelBackgroundEstimator
+
     """
-    Ring background method for cartesian coordinates.
+    def __init__(self, r_in, r_out_max, width, stepsize=None,
+                 threshold=0.1, theta=None, method='const. width'):
+
+        # default values
+        stepsize = stepsize if stepsize else 0.02 * u.deg
+        theta = theta if theta else 0.22 * u.deg
+
+        # input validation
+        if method not in ['const. width', 'const. r_in']:
+            raise ValueError("Not a valid adaptive ring method.")
+
+        self.parameters = dict(r_in=r_in, r_out_max=r_out_max, width=width,
+                               stepsize=stepsize, threshold=threshold,
+                               method=method, theta=theta)
+
+    def kernels(self, image):
+        """
+        Ring kernels according to the specified method.
+
+        Parameters
+        ----------
+        image : `~gammapy.image.SkyImage`
+            Sky image specifying the WCS information.
+
+        Returns
+        -------
+        kernels : list
+            List of `~astropy.convolution.Ring2DKernel`
+        """
+        p = self.parameters
+
+        scale = image.wcs_pixel_scale()[0]
```

Comment:
Maybe pass images to `__init__`?
Then the pix scale and number of kernels and their parameters would be known and could e.g. be printed in the class `__str__`.

Really objects like this Estimator are supposed to be instantiated, run and results read off.
Then if you need to run with different parameters or a new set of images, you just make a new one, no big deal.